Mage.Sets/src/mage/cards/g/GiantAlbatross.java

```diff
@@ -0,0 +1,105 @@
+package mage.cards.g;
+
+import java.util.List;
+import java.util.UUID;
+import mage.MageInt;
+import mage.MageObjectReference;
+import mage.abilities.Ability;
+import mage.abilities.common.DiesTriggeredAbility;
+import mage.abilities.costs.Cost;
+import mage.abilities.costs.common.PayLifeCost;
+import mage.abilities.costs.mana.ManaCostsImpl;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.DoIfCostPaid;
+import mage.cards.Card;
+import mage.constants.Outcome;
+import mage.constants.SubType;
+import mage.abilities.keyword.FlyingAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.filter.StaticFilters;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.players.Player;
+import mage.watchers.Watcher;
+import mage.watchers.common.DealtDamageToWatcher;
+import mage.watchers.common.PlayerDamagedBySourceWatcher;
+
+/**
+ *
+ * @author noahg
+ */
+public final class GiantAlbatross extends CardImpl {
+
+    public GiantAlbatross(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{1}{U}");
+        
+        this.subtype.add(SubType.BIRD);
+        this.power = new MageInt(1);
+        this.toughness = new MageInt(1);
+
+        // Flying
+        this.addAbility(FlyingAbility.getInstance());
+
+        // When Giant Albatross dies, you may pay {1}{U}. If you do, for each creature that dealt damage to Giant Albatross this turn, destroy that creature unless its controller pays 2 life. A creature destroyed this way can't be regenerated.
+        Ability ability = new DiesTriggeredAbility(new DoIfCostPaid(new GiantAlbatrossEffect(), new ManaCostsImpl("{1}{U}")));
+        DealtDamageToWatcher watcher = new DealtDamageToWatcher();
+        watcher.setSourceId(this.objectId);
```

Comment:
So the problem with using this watcher is that the ability can't be copied. If you were to clone this creature, the ability wouldn't work since the watcher wouldn't be watching its events. Take a look at Grothama, the Devourer.

Reply:
True, avoid to use watchers with scope Card. Always try to create game watchers if possible.

Reply:
And you can use permanent.getDealtDamageByThisTurn to get a list of objects that dealt damage to the permanent this turn.